src/test/java/org/apache/commons/io/input/UnsynchronizedByteArrayInputStreamTest.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.io.input;
+
+import org.junit.jupiter.api.Test;
+
+import static org.apache.commons.io.input.UnsynchronizedByteArrayInputStream.END_OF_STREAM;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Basic unit tests for the alternative ByteArrayInputStream implementation.
+ */
+public class UnsynchronizedByteArrayInputStreamTest {
```

Comment:
For new files, I like to keep method in alphabetical order, it makes it easier to find stuff.